### DISCUSSION TITLE
Why is `TOML.stringify()` output different for equivalent JSON input?

### DISCUSSION THREAD

**Discussion by Leftium** [LongTengDao/j-toml#27] [Q&A] [answered]
[`TOML.parse()`][1] seems to output a JSON object that somehow "remembers" the original TOML formatting. When the JSON is sent back to [`TOML.stringify()`][2] the original formatting is preserved. However when this JSON is "purified" by stringifying and parsing back to JSON, the formatting is no longer preserved.

How is j-toml "remembering" the format of the original TOML? How can I force j-toml to use a certain format for object literals that are added to the JSON?

More details: https://stackoverflow.com/q/76288348/117030

  [1]: https://github.com/LongTengDao/j-toml/tree/master/docs/English/#tomlparse
  [2]: https://github.com/LongTengDao/j-toml/tree/master/docs/English/#tomlstringify

**Comment by LongTengDao** [comment, marked as answer]
This library use `WeakMap` internal to remember object formats parsed by same library. For objects from `JSON.parse` or other way created, you may mark each object by helper functions provided by the library, because I didn't find a better way to know what coder want.

See details in docs: <https://github.com/LongTengDao/j-toml/tree/master/docs/English#tomlsection-tomlinline-tomlmultiline-tomlmultilinearray-tomlmultilinebasic-tomlbasic-tomlliteral-tomlcommentfor-tomlcommentforthis-tomlissection-tomlisinline>
